### PR TITLE
ci: Fix S3 Indexer Workflow.

### PR DESCRIPTION
**Pull request by tippmar-nr**
Thank you for submitting a pull request.  Please review our [contributing guidelines](/CONTRIBUTING.md) and [code of conduct](https://opensource.newrelic.com/code-of-conduct/).

## Description

The workflow that indexes files on the download site got broken with recent security best practice changes. This fixes the issue. 

# Author Checklist
- [ ] Unit tests, Integration tests, and Unbounded tests completed
- [ ] Performance testing completed with satisfactory results (if required)

# Reviewer Checklist
- [ ] Perform code review
- [ ] Pull request was adequately tested (new/existing tests, performance tests)

### CODE REVIEW

File: .github/workflows/deploy_agent.yml [hunk 1]
```diff
@@ -137,7 +137,7 @@ jobs:
     uses: ./.github/workflows/build_download_site_index_files.yml
     strategy:
       matrix:
-        prefix: ["dot_net_agent/latest_release", "dot_net_agent/previous_releases/${{ github.event.inputs.agent_version }}"]
+        prefix: ["dot_net_agent/latest_release", "dot_net_agent/previous_releases"]
```
[the remark this answers is not in the document]
Comment (tippmar-nr) [reply .github/workflows/deploy_agent.yml:140]: It's recursive and doesn't have any options to tell it not to. So the more releases we publish, the slower it'll get...

[the remark this answers is not in the document]
Comment (chynesNR) [reply .github/workflows/deploy_agent.yml:140]: Could we pass the Action a parameter for just the current version vs all versions? Or do we have to reindex every release?

File: .github/workflows/build_download_site_index_files.yml [hunk 2]
```diff
@@ -53,11 +54,6 @@ jobs:
     container:
       image: ghcr.io/newrelic/s3indexer
     steps:
-      - name: Harden Runner
```
Comment (nrcventura) [inline .github/workflows/build_download_site_index_files.yml:56]: Did the Harden Runner step cause a problem? Or was the main problem, the missing permission?

File: .github/workflows/deploy_agent.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (tippmar-nr) [reply .github/workflows/deploy_agent.yml:140]: Part of the problem is that it needs to re-index the `previous_releases` "folder" as well as the folder for the previous version - it wasn't correctly re-indexing the `previous_releases` folder even on prior successful runs, so it looked like 10.8.0 and 10.9.0 weren't even available (even though you could navigate directly to their respective folders).

File: .github/workflows/build_download_site_index_files.yml (hunk 2 above)
Comment (tippmar-nr) [reply .github/workflows/build_download_site_index_files.yml:56]: `harden-runner` doesn't support being run inside a docker container in a Github action so I removed it.

Comment (tippmar-nr) [reply .github/workflows/build_download_site_index_files.yml:56]: https://github.com/newrelic/newrelic-dotnet-agent/actions/runs/4671862007/jobs/8273364090#step:3:10

**Review by nrcventura — APPROVED** [review]

**Review by chynesNR — APPROVED** [review]

**Pull request merged**